fix(semantic): params in `export default (function() {})` flagged as `SymbolFlags::Export`

Dunqing opened the pull request:
The code changes in `builder.rs` optimize the export flag check in the `SemanticBuilder` implementation. Instead of setting the export flag to `true` for all cases except `ExportDefaultDeclarationKind::ClassDeclaration(ref class)`, the flag is now set to `true` only for `ExportDefaultDeclarationKind::ClassDeclaration` and `false` for all other cases. This change improves the efficiency of the export flag check.

Dunqing commented:
* **#4480** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4480?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

DonIsaac commented:
How about this case?
```ts
export default function foo() {
}
```
will `foo` still be marked as exported?

DonIsaac commented:
Yeah, this is breaking default exports for interfaces an enums
```ts
export default interface Foo {}
```

Dunqing commented:
> Yeah, this is breaking default exports for interfaces.
> 
> ```ts
> export default interface Foo {}
> ```

Ah, Thanks you found this. I also noticed we even don't have `Foo` symbol. 😅

Dunqing committed:
fix(semantic): params in `export default (function() {})` flagged as `SymbolFlags::Export` (#4480)

The code changes in `builder.rs` optimize the export flag check in the `SemanticBuilder` implementation. Instead of setting the export flag to `true` for all cases except `ExportDefaultDeclarationKind::ClassDeclaration(ref class)`, the flag is now set to `true` only for `ExportDefaultDeclarationKind::ClassDeclaration` and `false` for all other cases. This change improves the efficiency of the export flag check.
